pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,159 @@
+{ lib
+, fetchFromGitHub
+, fetchgit
+, crystal
+, libadwaita
+, gtk4
+, gtk3
+, glib
+, gobject-introspection
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+
+crystal.buildCrystalPackage rec {
+  pname = "Collision";
+  version = "3.5.0";
+
+  # project source
+  src = fetchFromGitHub {
+    owner = "GeopJr";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    name = pname;
```

Comment:
As much as I disagree with the consensus and would favor your way, we do not give sources a `name` in `nixpkgs`.

```suggestion
```

Reply:
Removed in https://github.com/NixOS/nixpkgs/pull/242681/commits/1de87e0e063a584308d57a83df37c752e7968fcb